TESTS/mbed_drivers/crc/main.cpp

```diff
@@ -0,0 +1,122 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "utest/utest.h"
+#include "unity/unity.h"
+#include "greentea-client/test_env.h"
+
+#include "mbed.h"
+
+using namespace utest::v1;
+
+void test_supported_polynomials()
+{
+    char  test[] = "123456789";
+    uint32_t crc;
+
+    {
+        MbedCRC<POLY_8BIT_CCITT, 8> ct;
+        TEST_ASSERT_EQUAL(0, ct.compute((void *)test, strlen((const char*)test), &crc));
+        TEST_ASSERT_EQUAL(0xF4, crc);
+    }
+    {
+        MbedCRC<POLY_16BIT_CCITT, 16> ct;
+        TEST_ASSERT_EQUAL(0, ct.compute((void *)test, strlen((const char*)test), &crc));
+        TEST_ASSERT_EQUAL(0x29B1, crc);
+    }
+    {
+        MbedCRC<POLY_16BIT_IBM, 16> ct;
+        TEST_ASSERT_EQUAL(0, ct.compute((void *)test, strlen((const char*)test), &crc));
+        TEST_ASSERT_EQUAL(0xBB3D, crc);
+    }
+    {
+        MbedCRC<POLY_32BIT_ANSI, 32> ct;
+        TEST_ASSERT_EQUAL(0, ct.compute((void *)test, strlen((const char*)test), &crc));
+        TEST_ASSERT_EQUAL(0xCBF43926, crc);
+    }
+}
+
+void test_partial_crc()
+{
+    char  test[] = "123456789";
+    uint32_t crc;
+    {
+        MbedCRC<POLY_16BIT_CCITT, 16> ct;
+        TEST_ASSERT_EQUAL(0, ct.compute_partial_start(&crc));
+        TEST_ASSERT_EQUAL(0, ct.compute_partial((void *)&test, 4, &crc));
+        TEST_ASSERT_EQUAL(0, ct.compute_partial((void *)&test[4], 5, &crc));
+        TEST_ASSERT_EQUAL(0, ct.compute_partial_stop(&crc));
+
+        TEST_ASSERT_EQUAL(0x29B1, crc);
+    }
+}
+
+void test_sd_crc()
+{
+    MbedCRC<POLY_7BIT_SD, 7> crc7;
+    uint32_t crc;
+    char test[512];
+
+    test[0] = 0x40;
+    test[1] = 0x00;
+    test[2] = 0x00;
+    test[3] = 0x00;
+    test[4] = 0x00;
+    TEST_ASSERT_EQUAL(0, crc7.compute((void *)test, 5, &crc));
+    crc = (crc | 0x1 ) & 0xFF;
+    TEST_ASSERT_EQUAL(0x95, crc);
+
+    test[0] = 0x48;
+    test[1] = 0x00;
+    test[2] = 0x00;
+    test[3] = 0x01;
+    test[4] = 0xAA;
+    TEST_ASSERT_EQUAL(0, crc7.compute((void *)test, 5, &crc));
+    crc = (crc | 0x1 ) & 0xFF;
+    TEST_ASSERT_EQUAL(0x87, crc);
+
+    test[0] = 0x51;
+    test[1] = 0x00;
+    test[2] = 0x00;
+    test[3] = 0x00;
+    test[4] = 0x00;
+    TEST_ASSERT_EQUAL(0, crc7.compute((void *)test, 5, &crc));
+    crc = (crc | 0x1 ) & 0xFF;
+    TEST_ASSERT_EQUAL(0x55, crc);
+
+    MbedCRC<POLY_16BIT_CCITT, 16> crc16(0, 0, false, false);
+    memset(test, 0xFF, 512);
+    TEST_ASSERT_EQUAL(0, crc16.compute((void *)test, 512, &crc));
+    TEST_ASSERT_EQUAL(0x7FA1, crc);
+}
+
+Case cases[] = {
+    Case("Test supported polynomials", test_supported_polynomials),
```

Comment:
Interesting. So is POLY_7BIT_SD not considered a supported polynomial?

Reply:
It is supported, but a special polynomial used for SD driver, i should rename that to standard supported polynomials